src/services/RetroGenerationService.ts

```diff
@@ -79,49 +79,61 @@ class RetroGenerationService {
+                                        const customDefectNotes = (customDefect.defectNotes) ? customDefect.defectNotes : "";
+                                        customDefects = customDefects + customDefect.referenceNumber + " " + customDefect.defectName + " " + customDefectNotes + "\r\n";
+                                    });
+                                }
+
+                                const certificateNumber = (!this.isTestTypeCoifWithAnnualTestOrCoifWithAnnualTestRetest(testType)) ? testType.certificateNumber :
+                                    testType.certificateNumber + " (Annual test), " + testType.secondaryCertificateNumber + " (COIF)";
+
+                                detailsTemplate.activity.value = (activity.activityType === "visit") ? ActivityType.TEST : ActivityType.WAIT_TIME;
+                                detailsTemplate.startTime.value = moment(testResult.testStartTimestamp).tz(TimeZone.LONDON).format("HH:mm:ss");
+                                detailsTemplate.finishTime.value = moment(testResult.testEndTimestamp).tz(TimeZone.LONDON).format("HH:mm:ss");
+                                detailsTemplate.vrm.value = (testResult.vehicleType === VEHICLE_TYPES.TRL) ? testResult.trailerId : testResult.vrm;
+                                detailsTemplate.chassisNumber.value = testResult.vin;
+                                detailsTemplate.testType.value = (testType.testCode).toUpperCase();
+                                detailsTemplate.seatsAndAxles.value = (testResult.vehicleType === VEHICLE_TYPES.PSV) ? testResult.numberOfSeats : testResult.noOfAxles;
+                                detailsTemplate.result.value = testType.testResult;
+                                detailsTemplate.certificateNumber.value = certificateNumber;
+                                detailsTemplate.expiryDate.value = testType.testExpiryDate ? moment(testType.testExpiryDate).tz(TimeZone.LONDON).format("DD/MM/YYYY") : "";
+                                detailsTemplate.preparerId.value = testResult.preparerId;
+                                detailsTemplate.failureAdvisoryItemsQAIComments.value = defects
+                                                                                      + reasonForAbandoning
+                                                                                      + additionalCommentsAbandon
+                                                                                      + LECNotes
+                                                                                      + "Additional test type notes: " + additionalTestTypeNotes + ";\r\n"
+                                                                                      + (testType.additionalNotesRecorded ? (testType.additionalNotesRecorded + ";") : "")
+                                                                                      + customDefects ? "\r\nCustom defects:\r\n" + customDefects : "";
```

Comment:
move `"\r\nCustom defects:\r\n" + customDefects` inside the `if (testType.customDefects)` to remove conditional check over here.

Reply:
fixed